src/ws/login.js

```diff
@@ -28,6 +28,19 @@
         return fmt.replace(fmt_re, function(m, x, y) { return args[x || y] || ""; });
     }
 
+    function select_language() {
+        var val = id("language").value;
+        if (!val)
+            val = "en-us";
```

Comment:
Not setting a language should IMHO mean "use the system default", not "use US English". If nothing is chosen, CockpitLang= should not be set at all. Again, this just mirros what we [currently do in the shell](https://github.com/cockpit-project/cockpit/blob/b444f9e03844/pkg/shell/base_index.js#L824), so not a regression, and fixing this in a follow-up is fine.

Reply:
That was a misunderstanding from my part: The language selector on login page is initialized with what the browser requests or what the cookie has, so it should already be correct. This is just a defensive fallback, and thus fine.